commit 8714215078a45552c3b35833260719c66ee316e2
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:29 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-port-force-edit.diff

diff --git a/src/port/port.tcl b/src/port/port.tcl
index 65e58a214..2ecf954db 100755
--- a/src/port/port.tcl
+++ b/src/port/port.tcl
@@ -3785,7 +3785,7 @@ proc action_portcmds { action portlist opts } {
                     set editor_var "ports_${action}_editor"
                     if {[dict exists $opts $editor_var]} {
                         set editor [join [dict get $opts $editor_var]]
-                    } else {
+                    } elseif {![macports::global_option_isset ports_force]} {
                         foreach ed { MP_EDITOR VISUAL EDITOR } {
                             if {[info exists env($ed)]} {
                                 set editor $env($ed)
@@ -3795,7 +3795,14 @@ proc action_portcmds { action portlist opts } {
                     }
 
                     # Use a reasonable canned default if no editor specified or set in env
-                    if { $editor eq "" } { set editor "/usr/bin/vi" }
+                    # or the user used the -f option to bypass any custom settings.
+                    if { $editor eq "" } {
+                        if {[file exists ${macports::prefix}/bin/vim]} {
+                            set editor "${macports::prefix}/bin/vim"
+                        } else {
+                            set editor "/usr/bin/vi"
+                        }
+                    }
 
                     # Invoke the editor
                     if {[catch {exec -ignorestderr >@stdout <@stdin {*}$editor $portfile} result]} {
